### PR TITLE
12.0 product uom migration

### PR DESCRIPTION
**Pull request by PCatinean**
Description of the issue/feature this PR addresses:

Migration to v12 of uom and product module

Current behavior before PR:

It doesn't load?

Desired behavior after PR is merged:

It loads if everything goes as planned :)

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by hmrodrigues** [comment]
I've a question regarding the base model migration.

How will we handle the admin user id?
In Odoo 12, the admin user now has the ID 2 and the root user (aka System) has the ID 1

I believe that the root user (ID 1) will not be a problem (the role is the same, but now has a different name and can't login (unable to set active=True, at least via ORM), but the admin user (now with id 2) can be, because the new admin will be a old normal user (or even a public one)

An example for this is the flow to set the [web.base.url](https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/res_users.py#L581) parameter. If a module implements something like this but without a try except block, the user will have issues with permissions.

**Comment by StefanRijnhart** [comment]
hmoridriques, can you create an issue to discuss this? It is not relevant for this particular PR

**Comment by pedrobaeza** [comment]
Can you rebase and solve existing conflicts with the new base? @mreficent do you have a replacement for this?

**Comment by MiquelRForgeFlow** [comment]
Yes and yes, but this PR also puts migration scripts for `product`, and `product` depends of `mail`. Until `mail` gets merged, I don't see the need to update this PR.

**Comment by pedrobaeza** [comment]
OK, I continue the escalation over dependencies and merging things. Any way, should we close this?

**Comment by MiquelRForgeFlow** [comment]
No need to close this. I will just update this branch in the following days :)

**Comment by MiquelRForgeFlow** [comment]
I doesn't understand the origin of the error in the CI

**Comment by pedrobaeza** [comment]
The problem here is that the module `uom` doesn't exist in v11, so no migration scripts are going to be processed for this module, as it's going being installed for the first time during migration. We have to do all the stuff in other module migration scripts (base?) or trick this. Investigating the best way right now...

**Comment by pedrobaeza** [comment]
And here we go! I have moved the `uom` renamings to `base` module, fix some things inside `product` migration scripts, and also fixed main attachment id of `mail` module method.

File: addons/mail/migrations/12.0.1.0/end-migration.py [hunk 1]
```diff
@@ -37,21 +37,25 @@ def fill_mail_thread_message_main_attachment_id(env):
...
+            prefetch_fields=False,
+        )
+        grouped_attachments = attachment_obj.read_group(
             [('res_model', '=', model)],
-            fields=['res_id', 'id:array_agg'],
+            fields=['res_id'],
             groupby=['res_id'])
-        attachs = [(a['res_id'], a['id']) for a in grouped_attachments]
+        attachs = {
+            a['res_id']: attachment_obj.search(a['__domain'])
```
Comment (MiquelRForgeFlow) [inline addons/mail/migrations/12.0.1.0/end-migration.py:48]: shouldn't we add `.with_context(active_test=False)` in this `search`? 🤔

Comment (pedrobaeza) [reply addons/mail/migrations/12.0.1.0/end-migration.py:48]: Has attachment an `active` field?

Comment (MiquelRForgeFlow) [reply addons/mail/migrations/12.0.1.0/end-migration.py:48]: Yes, it has.

Comment (MiquelRForgeFlow) [reply addons/mail/migrations/12.0.1.0/end-migration.py:48]: https://github.com/OCA/OpenUpgrade/pull/1678

Comment (davidtranhp) [reply addons/mail/migrations/12.0.1.0/end-migration.py:48]: attachment in Odoo 11 does not have `active` field.

File: odoo/addons/base/migrations/12.0.1.3/pre-migration.py [hunk 2]
```diff
@@ -1,12 +1,47 @@
...
+    ('product_uom', 'uom_uom'),
+    ('product_uom_categ', 'uom_category'),
+]
+
 xmlid_renames = [
     ('auth_signup.default_template_user', 'base.template_portal_user_id'),
     ('auth_signup.default_template_user_config',
      'base.default_template_user_config'),
+    ('product.group_uom', 'uom.group_uom'),
+    ('product.product_uom_categ_unit', 'uom.product_uom_categ_unit'),
```
Comment (cuongnmtm) [inline odoo/addons/base/migrations/12.0.1.3/pre-migration.py:24]: Missing  `('product.product_uom_categ_vol', 'uom.product_uom_categ_vol'),`

**Review by cuongnmtm — COMMENTED** [review]
Missing an item in `xmlid_renames`

File: odoo/addons/base/migrations/12.0.1.3/pre-migration.py (hunk 2 above)
Comment (pedrobaeza) [reply odoo/addons/base/migrations/12.0.1.3/pre-migration.py:24]: Can you point where in 11.0 code this record is defined?

Comment (davidtranhp) [reply odoo/addons/base/migrations/12.0.1.3/pre-migration.py:24]: Is it? https://github.com/odoo/odoo/blob/45bd7a58d5202d6b9387636317eb33efa569f458/addons/product/data/product_data.xml#L15

Comment (pedrobaeza) [reply odoo/addons/base/migrations/12.0.1.3/pre-migration.py:24]: I mean `product_uom_categ_vol` pointed by @cuongnmtm. That one doesn't exist...

Comment (davidtranhp) [reply odoo/addons/base/migrations/12.0.1.3/pre-migration.py:24]: This one: https://github.com/odoo/odoo/blob/45bd7a58d5202d6b9387636317eb33efa569f458/addons/product/data/product_data.xml#L27

Comment (pedrobaeza) [reply odoo/addons/base/migrations/12.0.1.3/pre-migration.py:24]: Is that v11?

Comment (davidtranhp) [reply odoo/addons/base/migrations/12.0.1.3/pre-migration.py:24]: Yes, that is. https://github.com/odoo/odoo/blob/11.0/addons/product/data/product_data.xml

Comment (pedrobaeza) [reply odoo/addons/base/migrations/12.0.1.3/pre-migration.py:24]: Thanks. Fixed in ab5e14825252bfd41d87c481d1318c148855b057

**Pull request merged**